packages/babel-plugin-transform-typescript/src/enum.ts

```diff
@@ -124,7 +125,29 @@ export function translateEnumValues(
           value = t.stringLiteral(constValue);
         }
       } else {
-        value = initializer;
+        const IdentifierVisitor: Visitor = {
+          Identifier(expr) {
+            if (seen.has(expr.node.name)) {
+              expr.replaceWith(
+                t.memberExpression(
+                  t.cloneNode(path.node.id),
+                  t.cloneNode(expr.node),
+                ),
+              );
+              expr.skip();
+            }
+          },
+          MemberExpression(expr) {
+            expr.skip();
+          },
+        };
+
+        const exprStmt = t.expressionStatement(initializer);
+
+        traverse(t.program([exprStmt]), IdentifierVisitor);
+
+        value = exprStmt.expression;
```

Comment:
Rather than creating a "fake" program in order to traverse this, you can:
1. Replace `return path.node.members.map(member => {` a few lines above with
    ```js
    return path.members.map(memberPath => {
    ```
    so that you have access to the `NodePath`s rather than to the raw nodes
2. Get the `initializerPath` with `memberPath.get("initializer")`
3. Traverse it with
   ```js
    if (initializerPath.isIdentifier()) {
      enumSelfReferenceVisitor.Identifier(initializerPath, { seen });
    } else {
      initializerPath.traverse(enumSelfReferenceVisitor, { seen });
    }
    ```

Reply:
Thanks, it helps a lot.

I've run into another problem.
After Visitor is used once, the internal `ReferencedIdentifier` function is not accessible again, which causes me to bring it up in the top scope.